src/test/java/jewellery/inventory/unit/service/ResourceInOrganizationServiceTest.java

```diff
@@ -0,0 +1,214 @@
+import jewellery.inventory.service.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ResourceInOrganizationServiceTest {
+
+  @InjectMocks private ResourceInOrganizationService resourceInOrganizationService;
+  @Mock private ResourceInOrganizationRepository resourceInOrganizationRepository;
+  @Mock private UserInOrganizationService userInOrganizationService;
+  @Mock private OrganizationService organizationService;
+  @Mock private ResourceService resourceService;
+  @Mock private ResourceInOrganizationMapper resourceInOrganizationMapper;
+
+  private Organization organization;
+  private Resource resource;
+  private ResourceInOrganizationRequestDto resourceInOrganizationRequestDto;
+  private ResourceInOrganization resourceInOrganization;
+  private static final BigDecimal QUANTITY = BigDecimal.ONE;
+  private static final BigDecimal NEGATIVE_QUANTITY = BigDecimal.valueOf(-5);
+  private static final BigDecimal BIG_QUANTITY = BigDecimal.valueOf(30);
+  private static final BigDecimal DEAL_PRICE = BigDecimal.TEN;
+
+  @BeforeEach
+  void setUp() {
+    organization = OrganizationTestHelper.getTestOrganization();
+    resource = ResourceTestHelper.getPearl();
+    resourceInOrganizationRequestDto =
+        ResourceInOrganizationTestHelper.createResourceInOrganizationRequestDto(
+            organization.getId(), resource.getId(), QUANTITY, DEAL_PRICE);
+    resourceInOrganization =
+        ResourceInOrganizationTestHelper.createResourceInOrganization(organization, resource);
+    organization.setResourceInOrganization(List.of(resourceInOrganization));
+  }
+
+  @Test
+  void testAddResourceToOrganizationShouldReturnNotFoundException() {
+    when(organizationService.getOrganization(any())).thenThrow(OrganizationNotFoundException.class);
+
+    Assertions.assertThrows(
+        OrganizationNotFoundException.class,
+        () ->
+            resourceInOrganizationService.addResourceToOrganization(
+                resourceInOrganizationRequestDto));
+  }
+
+  @Test
+  void testAddResourceToOrganizationShouldThrowMissingOrganizationPermissionException() {
+    when(organizationService.getOrganization(organization.getId())).thenReturn(organization);
+
+    doThrow(MissingOrganizationPermissionException.class)
+        .when(userInOrganizationService)
+        .validateCurrentUserPermission(organization, OrganizationPermission.ADD_RESOURCE_QUANTITY);
+
+    Assertions.assertThrows(
+        MissingOrganizationPermissionException.class,
+        () ->
+            resourceInOrganizationService.addResourceToOrganization(
+                resourceInOrganizationRequestDto));
+  }
+
+  @Test
+  void testAddResourceToOrganizationShouldThrowResourceNotFoundException() {
```

Comment:
I think we are duplicating this in the integration test

Reply:
Done!